src/App.jsx

```diff
@@ -14,45 +16,86 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App = () => (
-  <div className="section content">
-    <div className="buttons">
-      <button
-        type="button"
-        className="button is-info is-light"
-      >
-        Sort alphabetically
-      </button>
-
-      <button
-        type="button"
-        className="button is-success is-light"
-      >
-        Sort by length
-      </button>
-
-      <button
-        type="button"
-        className="button is-warning is-light"
-      >
-        Reverse
-      </button>
-
-      <button
-        type="button"
-        className="button is-danger is-light"
-      >
-        Reset
-      </button>
-    </div>
+const SORT_FIELD_LENGTH = 'length';
+const SORT_FIELD_ALPHABETICALLY = 'alphabetically';
+
+function getPreparedGoods(goods, sortField, isReversed) {
+  let preperedGoods = [...goods];
+
+  preperedGoods = preperedGoods
+    .sort((good1, good2) => {
+      switch (sortField) {
+        case SORT_FIELD_ALPHABETICALLY:
+          return good1.localeCompare(good2);
+        case SORT_FIELD_LENGTH:
+          return good1[sortField] - good2[sortField];
+        default: return 0;
+      }
+    });
+
+  if (isReversed) {
+    preperedGoods = preperedGoods.reverse();
+  }
+
+  return preperedGoods;
+}
+
+export const App = () => {
+  const [sortField, setSortField] = useState('');
+  const [isReversed, setIsReversed] = useState(false);
+  const visibleGoods = getPreparedGoods(goodsFromServer, sortField, isReversed);
 
-    <ul>
-      <li data-cy="Good">Dumplings</li>
-      <li data-cy="Good">Carrot</li>
-      <li data-cy="Good">Eggs</li>
-      <li data-cy="Good">Ice cream</li>
-      <li data-cy="Good">Apple</li>
-      <li data-cy="Good">...</li>
-    </ul>
-  </div>
-);
+  return (
+    <div className="section content">
+      <div className="buttons">
+        <button
+          type="button"
+          onClick={() => setSortField(SORT_FIELD_ALPHABETICALLY)}
+          className={cn('button', 'is-info', { 'is-light': sortField
+          !== SORT_FIELD_ALPHABETICALLY })}
+        >
+          Sort alphabetically
+        </button>
+
+        <button
+          type="button"
+          className={cn('button', 'is-success', { 'is-light': sortField
+          !== SORT_FIELD_LENGTH })}
+          onClick={() => setSortField(SORT_FIELD_LENGTH)}
+        >
+          Sort by length
+        </button>
+
+        <button
+          type="button"
+          className={cn(
+            'button',
+            'is-warning',
+            { 'is-light': !isReversed },
+          )}
+          onClick={() => setIsReversed(!isReversed)}
+        >
+          Reverse
+        </button>
+
+        {(sortField !== '' || isReversed !== false) && (
+          <button
+            type="button"
+            className="button is-danger is-light"
+            onClick={() => [setSortField(''), setIsReversed(false)]}
```

Comment:
```suggestion
            onClick={reset}
```
Consider moving this logic from jsx